src/main/kotlin/org/kamiblue/client/util/MovementUtils.kt

```diff
@@ -19,6 +20,7 @@ object MovementUtils {
     val Entity.isMoving get() = speed > 0.0001
     val Entity.speed get() = hypot(motionX, motionZ)
     val Entity.realSpeed get() = hypot(posX - prevPosX, posZ - prevPosZ)
+    val Entity.isInOrAboveLiquid get() =  world.containsAnyLiquid(entityBoundingBox.grow(0.0, -1.0, 0.0).shrink(0.001))
```

Comment:
I think this should be in EntityUtils. Also what is the shrink for?

Reply:
I'll move it to EntityUtils. The shrink is what Minecraft does to compensate for the variance of player positioning that can happen. I'm trying to stick to what MC calculates to avoid any easy detection methods.